music/src/core/visualizer.ts

```diff
@@ -456,3 +458,237 @@ export class PianoRollSVGVisualizer extends BaseVisualizer {
+      this.getScoreInfo(sequence), 
+      {
+        noteHeight: this.config.noteHeight,
+        noteSpacing: this.config.noteSpacing,
+        pixelsPerTimeStep: this.config.pixelsPerTimeStep,
+        noteRGB: this.config.noteRGB,
+        activeNoteRGB: this.config.activeNoteRGB,
+        defaultKey: config.defaultKey || 0,
+        scrollType: config.scrollType || ScrollType.PAGE,      
+      },
+      div
+    );
+    this.drawnNotes = sequence.notes.length;
+    this.clear();
+    this.redraw();
+  }
+
+  /**
+   * Clears and resets the visualizer object for further redraws from scratch.
+   */
+  protected clear() {
+    this.render.clear();
+  }
+
+  /**
+   * Redraws the entire `noteSequence` in a staff if no `activeNote` is given,
+   * highlighting on and off the appropriate notes otherwise. Should the 
+   * `noteSequence` had changed adding more notes at the end, calling this
+   * method again would complete the redrawing from the very last note it was
+   * drawn, maintaining the active note and the scroll position as they were. 
+   * This is handy for incremental compositions. Given the complexity of 
+   * adaption to a modified score, modifyied notes previously drawn will be
+   * ignored (you can always `clear()` and `redraw()` for a full redraw). 
+   * Please have in mind `mm.Player` does not have this incremental capability
+   * so, once the player had started, it will go on ignoring the changes. 
+   * 
+   * @param activeNote (Optional) If specified, this `Note` will be painted
+   * in the active color and there won't be an actual redrawing, but a 
+   * re-colouring of the involved note heads, accidentals, dots and ties 
+   * (activated and de-activated ones). Otherwise, all musical symbols which 
+   * were not processed yet will be drawn to complete the score.
+   * @param scrollIntoView (Optional) If specified and the active note to be 
+   * highlited is not visualized in the container DIV, the later will be 
+   * scrolled so that the note is viewed in the right place. This can be 
+   * altered by `AdvancedVisualizerConfig.scrollType`.
+   * @returns The x position of the highlighted active note relative to the 
+   * beginning of the SVG, or -1 if there wasn't any given active note. Useful
+   * for automatically advancing the visualization if needed.
+   */
+  public redraw(
+    activeNote?: NoteSequence.INote, 
+    scrollIntoView?: boolean 
+  ) : number {
+    if (this.drawnNotes !== this.noteSequence.notes.length) {
+      this.render.scoreInfo = this.getScoreInfo(this.noteSequence);
+    }
+    const activeNoteInfo = activeNote ? 
+      this.getNoteInfo(activeNote) : undefined;
+    return this.render.redraw(activeNoteInfo, scrollIntoView);
+  }
+
+  private isNoteInInstruments(note: NoteSequence.INote) : boolean {
+    if (note.instrument === undefined || this.instruments.length === 0) {
+      return true; // No instrument information in note means no filtering
```

Comment:
Super nit: The Google style guide says all comments need to end in a period (here and everywhere)

Reply:
Ok. No doubt it is super nit indeed :-D